Pod/Classes/URLSession+Extensions.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  NSURLSession+Extensions.swift
+//  Pods
+//
+//  Created by Matt Thomas on 3/17/16.
+//
+//
+import Foundation
+
+public typealias JSON = [AnyHashable: Any]
+
+public enum APIClientError: Error {
+    case cannotCreateURL
+    case badResponse(response: URLResponse)
+    case noData(response: URLResponse?)
+    case cannotSerializeToJSON(data: Data)
+    case resultJSONMissingKey(key: String, json: JSON)
+    
+    public var errorObject: NSError {
+        return NSError(domain: "MISLClientErrorDomain", code: errorInfo.code, userInfo: errorInfo.userInfo)
+    }
+    
+    fileprivate var errorInfo:(code: Int, userInfo: [AnyHashable: Any]?) {
+        switch self {
+            
+        case .cannotCreateURL:
+            return (code: 1, userInfo: nil)
+            
+        case .badResponse(let response):
+            return (code: 2, userInfo: ["response": response])
+            
+        case .noData(let response):
+            return (code: 3, userInfo: ["response": response ?? NSNull()])
+            
+        case .cannotSerializeToJSON(let data):
+            return (code: 4, userInfo: ["data": "\(data)"])
+            
+        case .resultJSONMissingKey(let key, let json):
+            return (code: 5, userInfo: [
+                "key": "\(key)",
+                "json": json
+                ])
+        }
+    }
+}
+
+public extension URLSession {
+    public func loadHTTP(request: URLRequest, completion: @escaping (_ result: Result<Data>) -> Void) {
+        self.load(request: request) { (response, result) in
+            // Because NSURLSessions "succeed" with any response, we need to check
+            // and see if the response is an HTTP 400-599 to see if it’s an HTTP failure.
+            if let httpResponse = response as? HTTPURLResponse , 400...599 ~= httpResponse.statusCode,
+                case .success = result {
+                    completion(Result(APIClientError.badResponse(response: httpResponse)))
+            }
+            else {
+                completion(result)
```

Comment:
Never mind, I see that you're getting a `ResultType` in the inner closure. Do you think you change those argument labels so its a bit more clear?